analytics.ts

```diff
@@ -0,0 +1,121 @@
+import * as Settings from '@taqueria/protocol/Settings';
+import { chain, map, resolve } from 'fluture';
+import { pipe } from 'https://deno.land/x/fun@v1.0.0/fns.ts';
+import { getMachineId } from 'https://deno.land/x/machine_id@v0.3.0/mod.ts';
+import type { DenoArgs } from './taqueria-types.ts';
+import * as utils from './taqueria-utils/taqueria-utils.ts';
+
+const {
+	doesPathExist,
+	readJsonFile,
+	writeJsonFile,
+	eager,
+} = utils.inject({
+	stdout: Deno.stdout,
+	stderr: Deno.stderr,
+});
+
+const consentPrompt =
+	'Allow Taqueria anonymously report usage statistics, in accordance with the privacy policy to help make Taqueria better? [y/yes] or [n/no]';
+const optInConfirmationPrompt =
+	'Allow Taqueria anonymously report usage statistics, in accordance with the privacy policy to help make Taqueria better? [y/yes] or [n/no]';
+const optOutConfirmationPrompt = 'Are you sure you want to turn off usage statistic reporting? [y/yes] or [n/no]';
+export const settingsFilePath = Deno.env.get('HOME') + '/.taq-settings.json';
+export const OPT_IN = 'opt_in';
+export const OPT_OUT = 'opt_out';
+
+export const optInAnalyticsFirstTime = () => createSettingsFileWithConsent(OPT_IN);
+
+export const optOutAnalyticsFirstTime = () => createSettingsFileWithConsent(OPT_OUT);
+
+const createSettingsFileWithConsent = (option: 'opt_in' | 'opt_out') =>
```

Comment:
Do you mind creating a type, `type consent = OPT_IN | OPT_OUT`?

This would be a better practice as if we ever need to add another value to the union then we would only need to update that in one place. Currently, we'd need to go to several places to add a new value.

It's possible we'd later add more granular consent options.

Reply:
-
After reviewing more, we can probably use our Settings type here instead of defining something new.

Reply:
I think I should go with your old suggestion of creating a new type because the Settings type may be extended to have more than Consent.